src/store/Controller.cc

```diff
@@ -203,7 +203,7 @@ Store::Controller::configure()
 
     // TODO: move this into a memory cache class when we have one
     const int64_t memMax = static_cast<int64_t>(min(Config.Store.maxInMemObjSize, Config.memMaxSize));
-    const int64_t disksMax = swapDir ? swapDir->maxObjectSize() : 0;
+    const int64_t disksMax = swapDir->maxObjectSize();
```

Comment:
> Removed unneccessary check for `swapDir` function in `Controller` since its assuming that `swapDir` will be valid at the begining of configure function.

This removal is not wrong, but since there is similar Controller code, it would be much better to remove _all_ such "unnecessary Controller::swapDir checks". Please do that in this PR. Each PR has a cost...

> Btw, does it make sense to check `swapDir` ptr at all?

On a low level, it does not, but the overall situation is a bit more complicated. Disk caching is an optional feature. Currently, Controller code is stuck between two correct designs:

1. Lack of a disk cache is best handled by a Disks object. The swapDir member is never nil.
2. Lack of a disk cache is best handled by Controller. The swapDir member may be nil. Disks class assumes that there is at least one cache_dir configured.

I prefer the first option because it should result in simpler, safer code. The current Controller code leans heavily towards that alternative. Addressing my change request will complete Controller migration to that design.

BTW, please use [PR description](https://github.com/measurement-factory/anubis/blob/master/README.md#pr-description) for the future commit message body. Use regular PR comments to ask questions. This approach both reduces the amount of manual work necessary to merge your PR and preserves the original discussion when/if PR description is edited.

P.S. The `swapDir` name itself is a bug. The correct name should have been something like `disks` or `cacheDirs`. Fixing that naming bug may not be worth the noise that fix will create and can be viewed as outside _this_ PR scope IMO.

Reply:
Got it. Thank you, will take into account all your comments